### DISCUSSION TITLE
MAC verification failed during PKCS12 import" error for iOS Code Signing with .p12 file

### DISCUSSION THREAD

**Discussion by kamilsouni** [codemagic-ci-cd/codemagic-docs#2805] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. On Apple Developer, I generated an Apple Distribution certificate.
2. I created an App Store provisioning profile, verified that it’s active, and linked it to the correct bundle ID com.example.xxxxx.
I attempted to generate a compatible .p12 file for import into Codemagic, but I’m encountering an error during the signing phase. Here are the methods I tried:
a. OpenSSL on Windows: I used various commands, including and excluding the -legacy option.
b. DigiCert Utility: I couldn’t link the private key with the certificate to create a .p12.
c. PowerShell: Attempted to directly associate the private key, without success.
d. In each case, I verified that the CERTIFICATE_PASSWORD variable matches the .p12 password, but the error persists.

### Expected results

The build should successfully import the .p12 file without error and properly sign the iOS app for distribution.

### Actual results

The build fails with the following error:
security: SecKeychainItemImport: MAC verification failed during PKCS12 import.
I suspect an incompatibility in the .p12 format generated by OpenSSL on Windows with Codemagic's requirements. I need guidance or an alternative method to generate a compatible .p12 file, or any insights on resolving the MAC verification error.

### Build id (optional)

672200f7ef10634b012b3056

**Comment by dam-ease** [comment, marked as answer]
Hey @kamilsouni
```
security: SecKeychainItemImport: MAC verification failed during PKCS12 import (wrong password?)
```
 usually mean that the provided .p12 certificate is somehow corrupt or password protected and the password is incorrect or you have not provided the certificate password.
I suggest you have your certificate without a password and re-upload the .p12 file.
Also you can consider using OpenSSL with the -legacy flag on Windows. This flag often helps avoid compatibility issues with the .p12 format on Windows systems

**Reply by benbucksch** [reply]
We had the same issue while building/signing Mac executables and adding `-legacy` to the OpenSSL p12 generation on Linux fixed it.
